drivers/modem/quectel-bg9x.c

```diff
@@ -1188,6 +1196,29 @@ static int modem_init(const struct device *dev)
 			(k_thread_entry_t) modem_rx,
 			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
 
+
```

Comment:
The IO Extender is a feature of the board it is designed into, not of a quectel modem driver. Therefor this should be either application code repository or a seperate io extender driver should be written for Zephyr. 

Besides I think enabling the power pins so that the driver gets battery voltage should be done before the quectel driver function `modem_init()` is called